pages/linux/gsettings.md

```diff
@@ -0,0 +1,32 @@
+# gsettings
+
+> Query and modify dconf settings with schema validation.
+> More information: <https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/using_the_desktop_environment_in_rhel_8/configuring-gnome-at-low-level_using-the-desktop-environment-in-rhel-8#using-gsettings-command_configuring-gnome-at-low-level>.
+
+- Display all (non-relocatable) schemas, keys, and values:
```

Comment:
Is there maybe an easier term or a way to describe `non-relocatable`?

Reply:
What I was trying to draw attention to is that `list-recursively` will not actually show *everything*.

What [the documentation](https://developer.gnome.org/gio/stable/GSettings.html) has to say about relocatable schemas is:

>A relocatable schema is one with no path attribute specified on its <schema> element.

[...]

>For example, a relocatable schema could be used to store geometry information for different windows in an application. If the schema ID was org.foo.MyApp.Window, it could be instantiated for paths /org/foo/MyApp/main/, /org/foo/MyApp/document-1/, /org/foo/MyApp/document-2/, etc.

[And elsewhere:](https://developer.gnome.org/gio/stable/gio-GSettingsSchema-GSettingsSchemaSource.html#g-settings-schema-get-path)

>Schemas may be single-instance or relocatable. Single-instance schemas correspond to exactly one set of keys in the backend database: those located at the path returned by this function.

>Relocatable schemas can be referenced by other schemas and can therefore describe multiple sets of keys at different locations. For relocatable schemas, this function will return `NULL`.

The main distinction seems to be that in order to operate on relocatable schemas, you need to know the GSettings schema id AND the dconf path.  `gsettings list-relocatable-schemas` can tell you the ids, but IDK how to get the paths where those schemas might be used.  [According to the discussion here](https://github.com/mate-desktop/mate-panel/issues/675), the `dconf-editor` GUI tool had a bear of a time trying to map the other way (from path to schema id).

It seems like most uses of the `gsettings` tool are, "paste this in your terminal to change such and such setting not exposed in the GUI," or new computer setup automation, which largely don't involve relocatable schemas.

Reply:
In most recent version, I have re-ordered the examples to put the subcommands with scripting uses, `set`/`get`/`reset`, come first and the "exploratory" ones, including `list-recursively`, are farther down.  This makes the stuff about non-relocatable vs relocatable schemas less prominent, so hopefully users are less likely to get the impression they need to understand what that means to use the tool.

But I do think it should be there somewhere, as a warning that `list-recursively` does not display everything.